core/src/main/java/org/apache/iceberg/BaseTable.java

```diff
@@ -48,6 +49,7 @@ public BaseTable(TableOperations ops, String name) {
   }
 
   public BaseTable(TableOperations ops, String name, MetricsReporter reporter) {
+    Preconditions.checkNotNull(reporter, "reporter cannot be null");
```

Comment:
```suggestion
    Preconditions.checkArgument(null != reporter, Invalid metrics reporter: null");
```

Reply:
it seems the codebase is generally OK using `checkNotNull` for null checks

```
$ git grep 'Preconditions.checkNotNull' | wc -l
392
```

are you saying this is passé?

Reply:
> Also we generally use `Preconditions.checkArgument` more than `Preconditions.checkNotNull`:

do we use it commonly to check for nullness?

i think in the code i've contributed so far to iceberg i used cNN and don't remember being turned around to use cA

Reply:
yeah as Piotr suggested, it seems like there are multiple places we already use cNN for this, so I am fine with continuing to use that. @nastra what do you think?

Reply:
the reason why I made the suggestion is mainly because the error msg should be updated to `Invalid metrics reporter: null`. Additionally I checked the code around table scans and scan metrics. Since those places all use  `Preconditions.checkArgument` it seems more appropriate to use the same here IMO.

Reply:
+1 for @nastra's comment about the error message.

Whether we use `checkNotNull` or `checkArgument` doesn't really matter because both are `ArgumentException`. The problem with this is that it makes an unnecessary change to have an error message that doesn't fit with the conventions used elsewhere.